packages/hop-node/src/db/GasPricesDb.ts

```diff
@@ -0,0 +1,36 @@
+import BaseDb from './BaseDb'
+import nearest from 'nearest-date'
+import { BigNumber } from 'ethers'
+import { DateTime } from 'luxon'
+import { normalizeDbItem } from './utils'
+
+export type GasPrice = {
+  chain?: string
+  timestamp?: number
+  gasPrice?: BigNumber
+}
+
+class GasPricesDb extends BaseDb {
+  async update (key: string, data: Partial<GasPrice>) {
+    return super.update(key, data)
+  }
+
+  async getNearest (chain: string, timestampSeconds: number): Promise<GasPrice> {
+    const keys = await this.getKeys()
+    const items = await Promise.all(
+      keys.map((key: string) => {
+        return this.getById(key)
+      })
+    )
+
+    const target = DateTime.fromSeconds(timestampSeconds)
+    const dates = items.map(item => DateTime.fromSeconds(item.timestamp).toJSDate())
```

Comment:
1. As we have it, `timestampSeconds` is in seconds and `item.timestamp` is in ms. I think they should be the same unit since we process them with the same logic (`DateTime.fromSeconds(x).toJSDate()`).
2. I don't know the best way, but I think we should be using `new DateTime()` instead of `DateTime.fromSeconds()`. Not 100% sure though.


```
var luxon = require("luxon")

const timeSec = 1632174913
const timeMs = 1632174913000

console.log(new luxon.DateTime(timeSec))
console.log(new luxon.DateTime(timeMs))

console.log(luxon.DateTime.fromSeconds(timeSec))
console.log(luxon.DateTime.fromSeconds(timeMs))

console.log(luxon.DateTime.fromSeconds(timeSec).toJSDate())
console.log(luxon.DateTime.fromSeconds(timeMs).toJSDate())


DateTime {ts: 1632178046986, _zone: SystemZone {}, …}
DateTime {ts: 1632178047018, _zone: SystemZone {}, …}
DateTime {ts: 1632174913000, _zone: SystemZone {}, …}
DateTime {ts: 1632174913000000, _zone: SystemZone {}, …}
Mon Sep 20 2021 14:55:13 GMT-0700 (Pacific Daylight Time)
Wed Jul 25 53691 01:16:40 GMT-0700 (Pacific Daylight Time)
```

Reply:
Good point, will make it be the time format